Sources/Core/AST/Stmt/ConditionalCompilationStmt.swift

```diff
@@ -0,0 +1,89 @@
+/// A conditional-compilation statement.
+public struct ConditionalCompilationStmt: Stmt {
+
+  public enum VersionComparison: Codable {
```

Comment:
```suggestion
  /// A predicate comparing compiler versions.
  public enum CompilerVersionComparison: Codable {
```
I _think_ you're talking about compiler version, which shows that being more specific will make that type clearer.

Reply:
Looking at the use of this type, I wonder if it wouldn't make more sense to bundle the predicate with the compiler version. The version would be a payload of an enum:
```
/// A predicate on a compiler version.
public enum CompilerVersionPredicate: Codable {

  /// Matches any compiler version greater than or equal to the payload.
  case greaterOrEqual(CompilerVersion)

  /// Matches any compiler version strictly inferior to the payload.
  case less(CompilerVersion)

}
```

Reply:
Ah, I see that you also want to use this type for language versions. Well, then I think the abstraction we want is a predicate over some expression of a semantic version.